app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,24 @@
 class PostsController < ApplicationController
   def index; end
 
-  def show; end
+  def show
+    @comment = Comment.new
+    @like = Like.new
+    @user = current_user
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author = current_user
+    @post.save
+    redirect_to "/users/#{current_user.id}/posts/#{@post.id}"
```

Comment:
- [ OPTIONAL ] I highly recommend here to add a condition to your case if else case so if  it is created so you will redirect to the post  but if it is not so it should redirect to the main page so you could do somthing like this 👍 

## Kindly do it in your comment create as well 💯💯
` def create

      @post = @current_user.posts.new(post_params)
      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end
`